pkg/workerpool/pool_impl.go

```diff
@@ -280,9 +280,6 @@ func (w *worker) run(ctx context.Context) error {
 		case <-ctx.Done():
 			return errors.Trace(ctx.Err())
 		case task := <-w.taskCh:
-			if task == nil {
-				return cerrors.ErrWorkerPoolEmptyTask.GenWithStackByArgs()
-			}
```

Comment:
Should we check empty task here?

Reply:
I don't think it necessary as task is used only in this file, and it never be empty.